src/test/java/uk/gov/pay/ledger/pact/CaptureConfirmedEventQueueContractTest.java

```diff
@@ -0,0 +1,102 @@
+import uk.gov.pay.ledger.util.fixture.QueueEventFixture;
+
+import java.time.ZonedDateTime;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+import static io.dropwizard.testing.ConfigOverride.config;
+import static org.awaitility.Awaitility.await;
+import static org.hamcrest.CoreMatchers.containsStringIgnoringCase;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+
+public class CaptureConfirmedEventQueueContractTest {
+    @Rule
+    public MessagePactProviderRule mockProvider = new MessagePactProviderRule(this);
+
+    @Rule
+    public AppWithPostgresAndSqsRule appRule = new AppWithPostgresAndSqsRule(
+            config("queueMessageReceiverConfig.backgroundProcessingEnabled", "true")
+    );
+
+    private byte[] currentMessage;
+    private String externalId = "externalId";
+    private ZonedDateTime eventDate = ZonedDateTime.parse("2018-03-12T16:25:01.123456Z");
+    private String gatewayAccountId = "gateway_account_id";
+
+    @Pact(provider = "connector", consumer = "ledger")
+    public MessagePact createCaptureConfirmedEventPact(MessagePactBuilder builder) {
+        String eventType = "CAPTURE_CONFIRMED";
+        QueueEventFixture captureConfirmedEvent = QueueEventFixture.aQueueEventFixture()
+                .withResourceExternalId(externalId)
+                .withEventDate(eventDate)
+                .withGatewayAccountId(gatewayAccountId)
+                .withEventType(eventType)
+                .withDefaultEventDataForEventType(eventType);
+
+        Map<String, String> metadata = new HashMap<String, String>();
+        metadata.put("contentType", "application/json");
+
+        PactDslJsonBody pactBody = captureConfirmedEvent.getAsPact();
+
+        return builder
+                .expectsToReceive("a capture confirmed message")
+                .withMetadata(metadata)
+                .withContent(pactBody)
+                .toPact();
+    }
+
+    @Test
+    @PactVerification({"connector"})
+    public void test() throws Exception {
+        appRule.getSqsClient().sendMessage(SqsTestDocker.getQueueUrl("event-queue"), new String(currentMessage));
+
+        TransactionDao transactionDao = new TransactionDao(appRule.getJdbi());
+        EventDao eventDao = appRule.getJdbi().onDemand(EventDao.class);
+        await().atMost(1, TimeUnit.SECONDS).until(
+                () -> {
+                    Optional<TransactionEntity> event;
+                    try {
+                        event = transactionDao.findTransactionByExternalId(externalId);
+                        return event.isPresent();
+
+                    } catch (NoSuchElementException e) {
+                        return false;
+                    }
+                }
+        );
+
+        var transaction = transactionDao.findTransactionByExternalId(externalId);
```

Comment:
Sorry, but can we type instead of var. It makes it difficult to follow `var` types

Reply:
Updated (both)